src/org/labkey/test/components/ui/search/EntityFieldFilterModal.java

```diff
@@ -4,32 +4,25 @@
 import org.labkey.test.Locator;
 import org.labkey.test.components.UpdatingComponent;
 import org.labkey.test.components.bootstrap.ModalDialog;
-import org.labkey.test.components.react.Tabs;
+import org.labkey.test.components.ui.grids.GridFilterModal;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 
 /**
  * Wraps 'labkey-ui-component' defined in <code>internal/components/search/EntityFieldFilterModal.tsx</code>
  */
-public class EntityFieldFilterModal extends ModalDialog
+public class EntityFieldFilterModal extends GridFilterModal
 {
-    private static final Locator listItem = Locator.byClass("list-group-item");
-
-    private final UpdatingComponent _linkedComponent;
-
-    protected EntityFieldFilterModal(WebDriver driver, UpdatingComponent linkedComponent)
+    public EntityFieldFilterModal(WebDriver driver, UpdatingComponent linkedComponent)
     {
-        super(new ModalDialog.ModalDialogFinder(driver));
-        _linkedComponent = linkedComponent;
+        super(driver, linkedComponent);
     }
 
     @Override
     protected void waitForReady(ModalDialog.ElementCache ec)
     {
-        super.waitForReady(ec);
-        getWrapper().shortWait().until(ExpectedConditions
-                .visibilityOf(listItem.findWhenNeeded(elementCache().querySelectionPanel)));
+        getWrapper().shortWait().until(ExpectedConditions.visibilityOf(listItem.findElementOrNull(elementCache().querySelectionPanel)));
```

Comment:
Need to use `findWhenNeeded` with `visibilityOf`
```suggestion
        getWrapper().shortWait().until(ExpectedConditions.visibilityOf(listItem.findWhenNeeded(elementCache().querySelectionPanel)));
```

Reply:
Done